evm/logic/context.py

```diff
@@ -193,3 +195,30 @@ def returndatacopy(computation):
 
 def sighash(computation):
     computation.stack.push(computation.msg.sig_hash)
+
+
+def paygas(computation):
+    gas_price = computation.stack.pop(type_hint=constants.UINT256)
+
+    # Only valid if (1) triggered in a top level call and
+    # (2) not been set already during this transaction execution
+    if computation.msg.depth == 0:
```

Comment:
The way this is written without an `else` clause this does nothing if triggered from `depth > 0`.  Is this intended?  Should it push `0` onto the stack if `depth > 0`?

Reply:
You are right. It should push `0` onto the stack if `depth > 0`.